src/main/java/budgetbuddy/model/account/Account.java

```diff
@@ -78,12 +107,14 @@ public boolean equals(Object other) {
 
     @Override
     public int hashCode() {
-        return Objects.hash(name, description, transactionList);
+        return Objects.hash(name, description, transactionList, balance);
     }
 
     @Override
     public String toString() {
         return getName().toString() + " (" + getDescription().toString() + ")";
     }
```

Comment:
I don't think the user will ever see the result of `toString()`, so up to you if you want to add the `balance` in it too.